controllers/resource/securityContext.go

```diff
@@ -0,0 +1,146 @@
+package resource
+
+import (
+	"maps"
+	"slices"
+	"strconv"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/cloudogu/cesapp-lib/core"
+	"github.com/cloudogu/k8s-dogu-operator/v3/api/v2"
+)
+
+func NewSecurityContextGenerator() *SecurityContextGenerator {
+	return &SecurityContextGenerator{}
+}
+
+type SecurityContextGenerator struct{}
+
+func (s *SecurityContextGenerator) Generate(dogu *core.Dogu, doguResource *v2.Dogu) (*corev1.PodSecurityContext, *corev1.SecurityContext) {
+	runAsNonRoot := isRunAsNonRoot(dogu, doguResource)
+	seLinuxOptions := seLinuxOptions(doguResource.Spec.Security.SELinuxOptions)
+	appArmorProfile := appArmorProfile(doguResource.Spec.Security.AppArmorProfile)
+	seccompProfile := seccompProfile(doguResource.Spec.Security.SeccompProfile)
+
+	readOnlyRootFS := isReadOnlyRootFS(dogu, doguResource)
+	// We never want those to be true and don't respect the dogu descriptor's privileged flag which is deprecated anyway.
```

Comment:
Nice :+1: This flag was only used by the registrator dogu which is no longer needed.